go/state/state_db_test.go

```diff
@@ -4143,6 +4143,71 @@ func TestSlotIdOrder(t *testing.T) {
 	}
 }
 
+func TestStateDB_SetTransientState_RollbackRemovesValue(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	mock := NewMockState(ctrl)
+	db := CreateStateDBUsing(mock)
+
+	s := db.Snapshot()
+	db.SetTransientState(address1, key1, val1)
+
+	db.RevertToSnapshot(s)
+	if got, want := db.GetTransientState(address1, key1), val0; got != want {
```

Comment:
you should compare for `var val common.Value`, I guess that `val0` is defined as an empty value, but it is just one of the test values, you should not rely on the fact it is zero.

Reply:
I added `valEmpty` as global to the testing data, since there are two places already where its checked.